code/modules/wiremod/components/utility/clock.dm

```diff
@@ -30,12 +38,25 @@
 	else
 		stop_process()
 
+	///minimum clock interval cant be smaller than COMP_CLOCK_DELAY
+	if(interval.value < 1)
+		interval.value = 1
+
 /obj/item/circuit_component/clock/Destroy()
 	stop_process()
 	return ..()
 
+/**
+ * Send signal only if sufficient ticks have elapsed
+ *
+ * since last signal
+ */
 /obj/item/circuit_component/clock/process(delta_time)
-	signal.set_output(COMPONENT_SIGNAL)
+	if(delta_last_signal >= interval.value*COMP_CLOCK_DELAY/(1 SECONDS))
+		signal.set_output(COMPONENT_SIGNAL)
+		delta_last_signal = COMP_CLOCK_DELAY/(1 SECONDS)
+	else
+		delta_last_signal = delta_last_signal + delta_time
```

Comment:
this is somewhat convoluted, can you just use a counter instead? That would be a lot more simple